app/models/manageiq/providers/oracle/infra_manager/refresh_parser.rb

```diff
@@ -0,0 +1,505 @@
+module ManageIQ::Providers::Oracle::InfraManager::RefreshParser
+  #
+  # EMS Inventory Parsing
+  #
+
+  def self.service_to_hashes(service)
+    uids = {}
+    result = {:uid_lookup => uids}
+
+    result[:storages], uids[:storages] = repository_inv_to_hashes(service.repositories)
+    result[:clusters], uids[:clusters], result[:resource_pools] = cluster_inv_to_hashes(service.server_pools)
+    result[:hosts], uids[:hosts], uids[:lans], uids[:switches], uids[:guest_devices], uids[:scsi_luns] = server_inv_to_hashes(service.servers, uids[:clusters])
+    result[:vms], uids[:vms] = vm_inv_to_hashes(service.vms, uids[:storages], uids[:clusters], uids[:hosts], uids[:lans])
+    result[:folders] = datacenter_inv_to_hashes(uids[:clusters], uids[:vms], uids[:storages], uids[:hosts])
+
+    # Link up the root folder
+    result[:ems_root] = result[:folders].first
+
+    result
+  end
+
+  def self.uri_to_ref(uri)
+    uri.split('/ovm/core/wsapi/rest').last
+  end
+
+  def self.repository_inv_to_hashes(inv)
+    result = []
+    result_uids = {:storage_id => {}}
+    return result, result_uids if inv.nil?
+
+    inv.each do |repository_inv|
+      mor = repository_inv.key
+
+      fs = repository_inv.file_system
+
+      new_result = {
+        :ems_ref             => uri_to_ref(repository_inv.uri),
+        :ems_ref_obj         => uri_to_ref(repository_inv.uri),
+        :name                => repository_inv.name,
+        :store_type          => 'ISCSI',
+        :total_space         => fs.size,
+        :free_space          => fs.free_size,
+        :uncommitted         => fs.free_size,
+        :multiplehostaccess  => true,
+        :location            => fs.path
+      }
+
+      result << new_result
+      result_uids[mor] = new_result
+      result_uids[:storage_id][repository_inv.key] = new_result
+    end
```

Comment:
It seems the provider written here was patterned from one of the older providers.  We have a much nicer pattern in the cloud stuff like Amazon and OpenStack, but I never got around to cleaning up the other ones and making them match.  I think this PR is fine as is, and we can clean *all* of it up together at some point.  I think @blomquisg was looking into that a little bit.